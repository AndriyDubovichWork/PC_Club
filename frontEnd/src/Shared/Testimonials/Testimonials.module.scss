.Reviews {
  width: 100%;
  background-color: #f8f8f8;

  text-align: center;

  .Content {
    width: 80vw;

    margin: 0 auto;

    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    .preTitle {
      font-weight: 600;

      margin-top: 10vh;
    }

    .Title {
      font-weight: 1600;
    }
    .subTitle {
      font-weight: 300;
      width: 50%;
    }
    .Testimonials {
      margin: 10vh 0;

      width: 100%;
      display: grid;
      justify-items: center;
      grid-template-columns: repeat(auto-fit, minmax(200px, 1fr));
      grid-row-gap: 2vh;
      grid-column-gap: 2vw;

      grid-auto-rows: 1fr;

      justify-content: space-around;
    }
  }
}
