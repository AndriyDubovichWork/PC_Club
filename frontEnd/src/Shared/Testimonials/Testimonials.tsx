import React from 'react';
import style from './Testimonials.module.scss';
import Testimonial from './Testimonial/Testimonial';

const Testimonials = () => {
  return (
    <div className={style.Reviews}>
      <div className={style.Content}>
        <h3 className={style.preTitle}>Reviewed by People</h3>
        <h1 className={style.Title}>Client's Testimonials</h1>
        <h4 className={style.subTitle}>
          Discover the positive impact we've made on the our clients by reading
          through their testimonials. Our clients have experienced our service
          and results, and they're eager to share their positive experiences
          with you.
        </h4>

        <div className={style.Testimonials}>
          <Testimonial
            name='Harry Potter'
            text='I had a great experience at PC Club! The staff was friendly and helpful. I highly recommend this store.'
            img={require('./assets/harry.jpg')}
          />

          <Testimonial
            img={require('./assets/Ron.jpg')}
            name='Ron Weasley'
            text="PC Club's great selection and knowledgeable staff make them my go-to for tech needs."
          />
        </div>
      </div>
    </div>
  );
};

export default Testimonials;
