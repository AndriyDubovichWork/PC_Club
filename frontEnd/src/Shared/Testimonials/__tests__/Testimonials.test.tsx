import { render, screen, cleanup } from '@testing-library/react';
import Testimonials from '../Testimonials';
import renderer from 'react-test-renderer';
import { MemoryRouter, Route, Routes } from 'react-router-dom';

afterEach(() => {
  cleanup();
});

test('Testimonials match snapshot', () => {
  const tree = renderer
    .create(
      <MemoryRouter initialEntries={['/PC_Club/']}>
        <Routes>
          <Route path='/PC_Club/' element={<Testimonials />} />
        </Routes>
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
test('Testimonials contains text', () => {
  render(
    <MemoryRouter initialEntries={['/PC_Club/']}>
      <Routes>
        <Route path='/PC_Club/' element={<Testimonials />} />
      </Routes>
    </MemoryRouter>
  );

  expect(screen.getByText('Reviewed by People')).toBeInTheDocument();
  expect(screen.getByText("Client's Testimonials")).toBeInTheDocument();
});
