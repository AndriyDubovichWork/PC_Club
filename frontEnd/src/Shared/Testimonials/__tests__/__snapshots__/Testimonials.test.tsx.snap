// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testimonials match snapshot 1`] = `
<div
  className="Reviews"
>
  <div
    className="Content"
  >
    <h3
      className="preTitle"
    >
      Reviewed by People
    </h3>
    <h1
      className="Title"
    >
      Client's Testimonials
    </h1>
    <h4
      className="subTitle"
    >
      Discover the positive impact we've made on the our clients by reading through their testimonials. Our clients have experienced our service and results, and they're eager to share their positive experiences with you.
    </h4>
    <div
      className="Testimonials"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-er2eq5-MuiPaper-root"
      >
        <div>
          <span
            className="Text"
          >
            "
            I had a great experience at PC Club! The staff was friendly and helpful. I highly recommend this store.
            "
          </span>
        </div>
        <div
          className="Author"
        >
          <img
            alt="testimonial"
            className="Img"
            loading="lazy"
            src="harry.jpg"
          />
          <h3>
            Harry Potter
          </h3>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-19pa9qe-MuiSvgIcon-root"
            data-testid="FormatQuoteIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 17h3l2-4V7H5v6h3zm8 0h3l2-4V7h-6v6h3z"
            />
          </svg>
        </div>
      </div>
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-er2eq5-MuiPaper-root"
      >
        <div>
          <span
            className="Text"
          >
            "
            PC Club's great selection and knowledgeable staff make them my go-to for tech needs.
            "
          </span>
        </div>
        <div
          className="Author"
        >
          <img
            alt="testimonial"
            className="Img"
            loading="lazy"
            src="Ron.jpg"
          />
          <h3>
            Ron Weasley
          </h3>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-19pa9qe-MuiSvgIcon-root"
            data-testid="FormatQuoteIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 17h3l2-4V7H5v6h3zm8 0h3l2-4V7h-6v6h3z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
